sklego/preprocessing/pandastransformers.py

```diff
@@ -42,17 +32,17 @@ def _nw_select_dtypes(df, include: str | list[str], exclude: str | list[str]):
         include = [include]
     if isinstance(exclude, str):
         exclude = [exclude]
-
-    include = include or ["string", "number", "bool", "category"]
-    exclude = exclude or []
-
-    feature_names = [
-        name
-        for name, dtype in df.schema.items()
-        if any(_nw_match_dtype(dtype, _include) for _include in include)
-        and not any(_nw_match_dtype(dtype, _exclude) for _exclude in exclude)
-    ]
-    return df.select(feature_names)
+    if include:
+        include = [_nw_match_dtype(x) for x in include]
+    else:
+        include = [nw.selectors.all()]
+    if exclude:
+        exclude = [_nw_match_dtype(x) for x in exclude]
+    else:
+        exclude = []
+    if exclude:
+        return df.select(functools.reduce(lambda x, y: x | y, include) - functools.reduce(lambda x, y: x | y, exclude))
+    return df.select(functools.reduce(lambda x, y: x | y, include))
```

Comment:
unfortunately Polars doesn't have an empty selector, else this could've been even simpler

Reply:
That would have been so much cleaner!